cmake/modules/kconfig.cmake

```diff
@@ -44,8 +48,26 @@ if(CONFIG_NCS_IS_VARIANT_IMAGE)
     # Additionally, convert primary slot dependencies to secondary slot dependencies.
     set(dotconfig_variant_content)
     foreach(line IN LISTS dotconfig_content)
+      dt_chosen(fw_to_relocate_property PROPERTY "fw-to-relocate")
       if("${line}" MATCHES "^CONFIG_FLASH_LOAD_OFFSET=.*$")
-        string(REGEX REPLACE "CONFIG_FLASH_LOAD_OFFSET=(.*)" "CONFIG_FLASH_LOAD_OFFSET=${code_partition_offset}" line ${line})
+        # Change the CONFIG_FLASH_LOAD_OFFSET value only if the fw_to_relocate_property is empty - meaning that the firmware is not being relocated.
+        if("${fw_to_relocate_property}" STREQUAL "")
+          string(REGEX REPLACE "CONFIG_FLASH_LOAD_OFFSET=(.*)" "CONFIG_FLASH_LOAD_OFFSET=${code_partition_offset}" line ${line})
+        endif()
+      endif()
+
+      # Change the CONFIG_BUILD_OUTPUT_ADJUST_LMA value only if the fw_to_relocate_property is not empty - meaning that the firmware is being relocated.
+      if(NOT "${fw_to_relocate_property}" STREQUAL "")
+        if("${line}" MATCHES "^CONFIG_BUILD_OUTPUT_ADJUST_LMA=.*$")
+
+          dt_partition_addr(fw_to_relocate_offset PATH "${fw_to_relocate_property}" REQUIRED)
+          dt_chosen(flash_controller PROPERTY "zephyr,flash")
+          dt_reg_addr(flash_base_addr PATH "${flash_controller}" REQUIRED)
```

Comment:
You may skip lines 64 and 65 by simply adding `ABSOLUTE` argument inside line 63.